crates/uv-resolver/src/lock.rs

```diff
@@ -610,8 +600,16 @@ impl Distribution {
 
     /// Add the [`AnnotatedDist`] as a dependency of the [`Distribution`].
     fn add_dependency(&mut self, annotated_dist: &AnnotatedDist, marker: Option<&MarkerTree>) {
-        self.dependencies
-            .push(Dependency::from_annotated_dist(annotated_dist, marker));
+        let new_dep = Dependency::from_annotated_dist(annotated_dist, marker);
+        for existing_dep in &mut self.dependencies {
+            if existing_dep.distribution_id == new_dep.distribution_id
+                && existing_dep.marker == new_dep.marker
```

Comment:
This makes `add_dependency` quadratic, right?

Reply:
I can't figure out a clean way around this.

Reply:
This is only called from `Lock::from_resolution_graph` and it's un-exported, so it should be fine to build up some intermediate state that replaces this loop with a `O(1)` or `O(logn)` check.

Reply:
Totally agree. I briefly tried but had some trouble making it work.

Reply:
For the large resolution transformers with all extras, this is the distribution of the number of iterations for the loop:

![image](https://github.com/user-attachments/assets/2a9d4bf3-2a68-44fd-bb7e-4c294165bac9)

We run the code inside the loop 2762 times.